plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/connection/OAuthDetailView.java

```diff
@@ -0,0 +1,364 @@
+			browser.setText(NLS.bind("<html><body><img style=\"display:block; margin-top: 50%; margin-left:auto; margin-right:auto\" alt=\"Embedded Image\" width=\"64\" height=\"64\" src=\"{0}\" /></body></html>",image));
+			setURL(url);
+			
+			return container;
+		}
+		
+		public void setURL(String url) {
+			if(StringUtils.isNotBlank(url)) {
+				this.url = url;
+				browser.setUrl(url);
+			}
+		}
+
+
+		@Override
+		public void aboutToRun(IJobChangeEvent event) {
+		}
+
+
+		@Override
+		public void awake(IJobChangeEvent event) {
+		}
+
+
+		@Override
+		public void done(final IJobChangeEvent event) {
+			if(event.getJob() instanceof AuthDetailsJob) {
+				Display.getDefault().asyncExec(new Runnable() {
+					@Override
+					public void run() {
+						AuthDetailsJob job = (AuthDetailsJob)event.getJob();
+						final IAuthorizationDetails details = job.getDetails();
+						if(details != null) {
+							setURL(details.getRequestTokenLink());
+						}else {
+							IStatus result = job.getResult();
+							showErrorDialog(getShell(),result.getException());
```

Comment:
Isnt it sufficient to have display#syncExec around #showErrorDialog or maybe also #setUrl() (is any other UI thread operation involved in the above code?